empress_gui.py

```diff
@@ -4,362 +4,819 @@
+        self.dup_input.set(1.00)
+        self.trans_input.set(1.00)
+        self.loss_input.set(1.00)
+        self.compute_reconciliations_btn.configure(state=tk.DISABLED)
+        self.view_solution_space_dropdown.configure(state=tk.DISABLED)
+        self.view_reconciliations_dropdown.configure(state=tk.DISABLED)
+        self.view_pvalue_histogram_btn.configure(state=tk.DISABLED)
+
+        # Reset the rest
+        self.dup_label.destroy()
+        self.dup_entry_box.destroy()
+        self.trans_label.destroy()
+        self.trans_entry_box.destroy()
+        self.loss_label.destroy()
+        self.loss_entry_box.destroy()
+
+        self.recon_MPRs_label.destroy()
+        self.num_MPRs_label.destroy()
+        self.recon_cospeci_label.destroy()
+        self.recon_dup_label.destroy()
+        self.recon_trans_label.destroy()
+        self.recon_loss_label.destroy()
+
+        self.num_cluster_input = tk.IntVar()
+        self.num_cluster_input.set(1)
+        self.num_cluster = None
+
+        self.recon_info_displayed = False
+        
+        if self.cost_space_window is not None and self.cost_space_window.winfo_exists() == 1:
+            self.cost_space_window.destroy()
+
+        if self.entire_space_window is not None and self.entire_space_window.winfo_exists() == 1:
+            self.entire_space_window.destroy()
+        
+        if self.set_num_cluster_window is not None and self.set_num_cluster_window.winfo_exists() == 1:
+            self.set_num_cluster_window.destroy()
+
+        if self.view_solution_space_window is not None and self.view_solution_space_window.winfo_exists() == 1:
+            self.view_solution_space_window.destroy()
+
+        if self.one_MPR_window is not None and self.one_MPR_window.winfo_exists() == 1:
+            self.one_MPR_window.destroy()
+
+        if self.view_reconciliations_window is not None and self.view_reconciliations_window.winfo_exists() == 1:
+            self.view_reconciliations_window.destroy()
+
+        if self.view_pvalue_histogram_window is not None and self.view_pvalue_histogram_window.winfo_exists() == 1:
+            self.view_pvalue_histogram_window.destroy()
+
+    def load_input_files(self, event):
+        """Load in two .nwk files for the host tree and parasite tree, and one .mapping file. Display the number of tips for 
+        the trees and a message to indicate the successful reading of the tips mapping.""" 
+        # Clicking on "Load host tree file" 
+        if self.load_files_var.get() == "Load host tree file":
+            # initialdir is set to be the current working directory
+            input_file = tk.filedialog.askopenfilename(initialdir=os.getcwd(), title="Select a host file")
+            if Path(input_file).suffix == '.nwk':
+                # Try to read in host tree file
+                self.recon_input.read_host(input_file)
+                self.host_file_path = None
+                if self.recon_input.host_dict is not None:
```

Comment:
Don't do this either. After reading the host_dict, assume that it's there correctly. It's the job of `ReconInput` functions to do the error checking for you and raise an Exception in case something goes wrong.

Reply:
ok